### PR TITLE
Add type definitions for Phaser

### PR DESCRIPTION
**Pull request by markogresak**
Copied from photonstorm/phaser repo.
Referring to issue photonstorm/phaser/issues/1724.

This is not my work, just adding it here for the convenience.

### CODE REVIEW

**Comment by vvakame** [comment]
*phaser/p2.d.ts*

check list

* [ ] is collect [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)?
  * https://www.npmjs.com/package/phaser
  * http://bower.io/search/?q=phaser
  * others?
* [ ] has a [test file](http://definitelytyped.org/guides/contributing.html#tests)? (phaser/p2-tests.ts or others)
* [ ] pass the Travic-CI test?

---

*phaser/phaser.comments.d.ts*

check list

* [ ] is collect [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)?
  * https://www.npmjs.com/package/phaser
  * http://bower.io/search/?q=phaser
  * others?
* [ ] has a [test file](http://definitelytyped.org/guides/contributing.html#tests)? (phaser/phaser.comments-tests.ts or others)
* [ ] pass the Travic-CI test?

---

*phaser/pixi.comments.d.ts*

check list

* [ ] is collect [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)?
  * https://www.npmjs.com/package/phaser
  * http://bower.io/search/?q=phaser
  * others?
* [ ] has a [test file](http://definitelytyped.org/guides/contributing.html#tests)? (phaser/pixi.comments-tests.ts or others)
* [ ] pass the Travic-CI test?

**Comment by vvakame** [comment]
can you create phaser/phaser.d.ts? it XX/XX.d.ts naming pattern is good for definition file manager. (e.g. tsd

**Comment by markogresak** [comment]
I've amended my commit, renaming files to remove `.comments` part. 

For the tests, I'm not sure what should be tested? I think testing each function would be a bit too much.
From what I found, phaser has two examples: [1](http://phaser.io/tutorials/how-to-use-phaser-with-typescript), [2](http://www.photonstorm.com/phaser/advanced-phaser-and-typescript-projects). I could include those two examples to assert that it compiles. But this would only tests phaser library, but not the p2 and pixi, upon which the library (and some developers) depend.

**Comment by vvakame** [comment]
@markogresak please read http://definitelytyped.org/guides/contributing.html#tests
test code is useful for reviewing and test backward compat about next pull request.

**Comment by markogresak** [comment]
I have read the guide, but I still haven't understood it fully. Therefore I asked the question in earlier comment.

**Comment by vvakame** [comment]
@basarat

File: phaser/p2.d.ts [hunk 1]
```diff
@@ -0,0 +1,950 @@
+// Type definitions for p2.js v0.6.0
+// Project: https://github.com/schteppe/p2.js/
```
Comment (basarat) [inline phaser/p2.d.ts]: shouldn't it be https://github.com/photonstorm/phaser ?

**Comment by basarat** [comment]
> Therefore I asked the question in earlier comment.

@markogresak would a reference please. Specifically would love to have at least a simple *test* to document how to use the definition. Feel free to ask questions about any difficulty :rose:

**Comment by markogresak** [comment]
I took the url form [phaser's definitions](https://github.com/photonstorm/phaser/blob/master/typescript/p2.d.ts#L2) so I guess it should be correct. 

Btw for some reason, the first time I've amended my commit, the phaser and pixi definitions were not added, therefore I had to amend it again to fix my mistake. 

I will try to put up an example to test at least some of the library's aspects, although I'm not yet sure how to test other libraries than phaser. I will try to ask for more help at phaser repo, because I'm just a beginner at phaser and I wanted to test out how it works with typescript. 
All I wanted to do is get these definitions on tsd, I apologize for doing a poor job on reading the docs.

**Comment by basarat** [comment]
> I took the url form phaser's definitions so I guess it should be correct.

I think this is the file you are looking for : https://github.com/photonstorm/phaser/blob/master/typescript/phaser.d.ts 

You would probably need to bring in definitions for `phaser` & `pixi` & `p2` based on https://github.com/photonstorm/phaser/blob/4b22f48c75eb08f46b3359181bad42ca5bce81b3/typescript/phaser.d.ts#L1-L2 

for `p2` and `pixi` would need to be moved into their appropriate directories. Prefer directory names based on `npm` (if those packages are indeed available on `npm`)

**Comment by markogresak** [comment]
As in photonstorm/phaser#2008, full coverage of Phaser would be a very difficult thing to do and since this repo requires all declarations to have test files, I'm closing this PR.

**Pull request closed without merging**